src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/WebRequestPSCmdlet.CoreClr.cs

```diff
@@ -408,6 +412,21 @@ protected override void StopProcessing()
 
         #region Helper Methods
 
+        private void ThrowNotSuccessStatusCodeError(HttpRequestMessage request, HttpResponseMessage response)
+        {
+            using(var reader = new StreamReader(StreamHelper.GetResponseStream(response)))
```

Comment:
wow, you've just found one more issue: IOException is not handled even if the response was ok, I thought it was handled somewhere in WebRequestPSCmdlet. I'm going to fix it as well.